pkg/controller/service/servicecontroller_test.go

```diff
@@ -296,3 +298,311 @@ func TestGetNodeConditionPredicate(t *testing.T) {
+		updateFn   func()                           //Manupulate the structure
+		srv        *v1.Service                      //Service component
+		expectedFn func(error, time.Duration) error //Error comparision function
+	}{
+		{
+			testName: "If updating a valid service",
+			key:      "validKey",
+			srv:      defaultExternalService(),
+			updateFn: func() {
+
+				controller, cloud, _ = newController()
+				controller.cache.getOrCreate("validKey")
+
+			},
+			expectedFn: func(err error, retryDuration time.Duration) error {
+
+				if err != nil {
+					return err
+				}
+				if retryDuration != doNotRetry {
+					return fmt.Errorf("retryDuration Expected=%v Obtained=%v", doNotRetry, retryDuration)
+				}
+				return nil
+			},
+		},
+	}
+
+	for _, tst := range testCases {
+		tst.updateFn()
+		srvCache := controller.cache.getOrCreate(tst.key)
+
+		obtErr, retryDuration := controller.processServiceUpdate(srvCache, tst.srv, tst.key)
+		if err := tst.expectedFn(obtErr, retryDuration); err != nil {
+			t.Errorf("%v processServiceUpdate() %v", tst.testName, err)
+		}
+	}
+
+}
+
+func TestSyncService(t *testing.T) {
+
+	var controller *ServiceController
+	var cloud *fakecloud.FakeCloud
+
+	testCases := []struct {
+		testName    string //Name of this test case
+		key         string //Name of the key we should be looking for
+		updateFn    func() //Function to manipulate the controller element to simulate error
+		expectedErr error  //syncService() only returns error
+	}{
+		{
+			testName: "if an invalid service name is synced",
+			key:      "invalid/key/string",
+			updateFn: func() {
+				controller, cloud, _ = newController()
+
+			},
+			expectedErr: fmt.Errorf("unexpected key format: %q", "invalid/key/string"),
+		},
+		/* We cannot open this test case as syncService(key) currently runtime.HandleError(err) and suppresses frequently occurring errors
+		{
+			testName: "if an invalid service is synced",
+			key: "somethingelse",
+			updateFn: func() {
+				controller, cloud, _ = newController()
+				srv := controller.cache.getOrCreate("external-balancer")
+				srv.state = defaultExternalService()
+			},
+			expectedErr: fmt.Errorf("Service somethingelse not in cache even though the watcher thought it was. Ignoring the deletion."),
+		},*/
+		{
+			testName: "if valid service",
+			key:      "external-balancer",
+			updateFn: func() {
+				controller, cloud, _ = newController()
+				srv := controller.cache.getOrCreate("external-balancer")
+				srv.state = defaultExternalService()
+			},
+			expectedErr: nil,
+		},
+	}
+
+	for _, tst := range testCases {
+		var printError bool
+		tst.updateFn()
+		obtainedErr := controller.syncService(tst.key)
+		if obtainedErr != tst.expectedErr {
+			printError = true
+			if obtainedErr != nil && tst.expectedErr != nil {
+				if obtainedErr.Error() == tst.expectedErr.Error() {
+					//Both are valid error objects but still their content is same, then not an error
```

Comment:
This sentence does not read well to me. Can you rephrase to clarify your intent? The comment might also be unnecessary.